pilot/pkg/model/service.go

```diff
@@ -107,8 +107,26 @@ type Service struct {
 	// MeshExternal (if true) indicates that the service is external to the mesh.
 	// These services are defined using Istio's ServiceEntry spec.
 	MeshExternal bool
+
+	// InternalTrafficPolicy specifies if the cluster internal traffic
+	// should be routed to all endpoints or node-local endpoints only.
+	// "Cluster" routes internal traffic to a Service to all endpoints.
+	// "Local" routes traffic to node-local endpoints only, traffic is
+	// dropped if no node-local endpoints are ready.
+	InternalTrafficPolicy *ServiceInternalTrafficPolicyType
```

Comment:
I don't think we need a pointer. Just make empty string equal the default behavior (ie "Cluster"). Pointers are hard to use, error prone, and have higher memory cost.

Really it could be a bool...

Reply:
> I don't think we need a pointer. Just make empty string equal the default behavior (ie "Cluster"). Pointers are hard to use, error prone, and have higher memory cost.
> Really it could be a bool...


Yes, it has been changed to boolean type